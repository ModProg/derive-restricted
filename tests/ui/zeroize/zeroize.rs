extern crate zeroize_ as zeroize;

use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where(Zeroize(test))]
struct InvalidOption<T>(PhantomData<T>);

#[derive_where(Zeroize(test = "test"))]
struct WrongOptionSyntax1<T>(PhantomData<T>);

#[derive_where(Zeroize("option"))]
struct WrongOptionSyntax2<T>(PhantomData<T>);

#[derive_where(Zeroize(crate(zeroize_)))]
struct WrongCrateSyntax<T>(PhantomData<T>);

#[derive_where(Zeroize(crate = struct Test))]
struct InvalidPath<T>(PhantomData<T>);

#[derive_where(Zeroize(crate = "struct Test"))]
struct InvalidPathDeprecated<T>(PhantomData<T>);

#[derive_where(Zeroize(crate = zeroize_, crate = zeroize_))]
struct DuplicateCrate<T>(PhantomData<T>);

#[derive_where(Zeroize(crate = ::zeroize))]
struct DefaultCrate<T>(PhantomData<T>);

fn main() {}
