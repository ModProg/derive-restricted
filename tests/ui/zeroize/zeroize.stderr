error: `Zeroize` doesn't support this option
 --> tests/ui/zeroize/zeroize.rs:7:24
  |
7 | #[derive_where(Zeroize(test))]
  |                        ^^^^

error: `Zeroize` doesn't support this option
  --> tests/ui/zeroize/zeroize.rs:10:24
   |
10 | #[derive_where(Zeroize(test = "test"))]
   |                        ^^^^

error: expected identifier
  --> tests/ui/zeroize/zeroize.rs:13:24
   |
13 | #[derive_where(Zeroize("option"))]
   |                        ^^^^^^^^

error: unexpected option syntax
  --> tests/ui/zeroize/zeroize.rs:16:24
   |
16 | #[derive_where(Zeroize(crate(zeroize_)))]
   |                        ^^^^^^^^^^^^^^^

error: expected an expression
  --> tests/ui/zeroize/zeroize.rs:19:32
   |
19 | #[derive_where(Zeroize(crate = struct Test))]
   |                                ^^^^^^

error: expected path, expected identifier, found keyword `struct`
  --> tests/ui/zeroize/zeroize.rs:22:32
   |
22 | #[derive_where(Zeroize(crate = "struct Test"))]
   |                                ^^^^^^^^^^^^^

error: duplicate `crate` option
  --> tests/ui/zeroize/zeroize.rs:25:42
   |
25 | #[derive_where(Zeroize(crate = zeroize_, crate = zeroize_))]
   |                                          ^^^^^^^^^^^^^^^^

error: unnecessary path qualification, `::zeroize` is used by default
  --> tests/ui/zeroize/zeroize.rs:28:32
   |
28 | #[derive_where(Zeroize(crate = ::zeroize))]
   |                                ^^^^^^^^^
